### ISSUE TITLE
installed_paths depend on PHP executable?

### ISSUE THREAD

**Issue by kkmuffme** [PHPCSStandards/composer-installer#216]
## Problem/Motivation

The loaded rulesets seems to be PHP-executing version specific, while composer itself is not.
My composer.json has `platform.php '8.3.0'` set.
My default PHP is 8.0

I install/update normally via `composer install` and see everything is correct:
>PHP CodeSniffer Config installed_paths set to ....,../../wp-coding-standards/wpcs

Then I run phpcbf with php 8.3 like: (my-ruleset.xml contains WordPress ruleset)
`/usr/bin/php83 /path/to/vendor/bin/phpcbf --standard=/path/to/my-ruleset.xml /other/path/file.php`

## Expected behaviour

no error

## Actual behavior

Error:
>Referenced sniff "WordPress" does not exist

## Fix

`/usr/bin/php83 -d xdebug.mode=off /usr/bin/composer install`

I get the same output as above 
>PHP CodeSniffer Config installed_paths set to ....,../../wp-coding-standards/wpcs

but then it works correctly.

## Proposed changes

The config should be loaded independently of the actually executing PHP script.
This is especially relevant in CI, where 1 PHP version might install packages for various executable PHP versions using `platform.php`.

## Environment

| Question                      | Answer
| ----------------------------- | -------
| OS                            | Linux
| PHP version                   | 8.3
| Composer PHPCS plugin version | 1.0.0
| Install type                  | Composer global

## Tested against `main` branch?
- [x] I have verified the issue still exists in the `main` branch.

**Comment by jrfnl** [comment]
@kkmuffme This issue is completely unclear to me. This plugin will use the PHP version used during the Composer install to run whatever commands it needs, but the end-result is a `CodeSniffer.conf` file which is not specific to any particular PHP version and the same can be said about running PHPCS, so I don't understand what you are trying to report/what is the perceived problem.

**Comment by jrfnl** [comment]
Cannot reproduce.

```json
{
	"name": "installes/216",
	"require-dev": {
		"squizlabs/php_codesniffer": "^3.8.0",
		"wp-coding-standards/wpcs": "^3.0.1"
	},
	"config": {
		"allow-plugins": {
			"dealerdirect/phpcodesniffer-composer-installer": true
		},
		"platform": {
			"php": "8.3.0"
		}
	}
}
```

```
$ composer install
PHP 8.0.30 (cli) (built: Sep  1 2023 14:15:38) ( ZTS Visual C++ 2019 x64 )
Copyright (c) The PHP Group
Zend Engine v4.0.30, Copyright (c) Zend Technologies
    with Xdebug v3.2.2, Copyright (c) 2002-2023, by Derick Rethans

Composer version 2.6.6 2023-12-08 18:32:26

No composer.lock file present. Updating dependencies to latest instead of installing from lock file. See https://getcomposer.org/install for more information.
Loading composer repositories with package information
Updating dependencies
Lock file operations: 5 installs, 0 updates, 0 removals
  - Locking dealerdirect/phpcodesniffer-composer-installer (v1.0.0)
  - Locking phpcsstandards/phpcsextra (1.2.1)
  - Locking phpcsstandards/phpcsutils (1.0.9)
  - Locking squizlabs/php_codesniffer (3.8.1)
  - Locking wp-coding-standards/wpcs (3.0.1)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 5 installs, 0 updates, 0 removals
  - Installing squizlabs/php_codesniffer (3.8.1): Extracting archive
  - Installing dealerdirect/phpcodesniffer-composer-installer (v1.0.0): Extracting archive
  - Installing phpcsstandards/phpcsutils (1.0.9): Extracting archive
  - Installing phpcsstandards/phpcsextra (1.2.1): Extracting archive
  - Installing wp-coding-standards/wpcs (3.0.1): Extracting archive
Generating autoload files
4 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
PHP CodeSniffer Config installed_paths set to ../../phpcsstandards/phpcsextra,../../phpcsstandards/phpcsutils,../../wp-coding-standards/wpcs
```
```
$ php -v
PHP 8.3.0 (cli) (built: Nov 21 2023 17:48:31) (ZTS Visual C++ 2019 x64)
Copyright (c) The PHP Group
Zend Engine v4.3.0, Copyright (c) Zend Technologies
    with Zend OPcache v8.3.0, Copyright (c), by Zend Technologies
    with Xdebug v3.3.0alpha3, Copyright (c) 2002-2023, by Derick Rethans
```
```
$ "vendor/bin/phpcs" -i
The installed coding standards are MySource, PEAR, PSR1, PSR2, PSR12, Squiz, Zend, Modernize, NormalizedArrays, Universal, PHPCSUtils, WordPress, WordPress-Core, WordPress-Docs and WordPress-Extra
```
```
$ "vendor/bin/phpcs" -ps ./test.php --standard=WordPress
E 1 / 1 (100%)

FILE: test.php
-----------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-----------------------------------------------------------------------------
 1 | ERROR | Missing file doc comment (Squiz.Commenting.FileComment.Missing)
-----------------------------------------------------------------------------

Time: 574ms; Memory: 12MB
```

**Comment by jrfnl** [comment]
@kkmuffme Only thing I can think of at this moment is that your custom ruleset contains an `<config...` setting overruling the installed paths or that something in your Composer `scripts` is screwing things up.

**Comment by jrfnl** [comment]
> in both your cases your "php" executable was the default one (despite being different versions)

No, I explicitly run with specific PHP version, just do so under the hood. Shouldn't make a difference anyhow (and it doesn't).

```
$ C:\wamp\bin\php\php8.3.0\php.exe "vendor/bin/phpcs" -ps ./test.php --standard=WordPress --basepath=.
E 1 / 1 (100%)

FILE: test.php
-----------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-----------------------------------------------------------------------------
 1 | ERROR | Missing file doc comment (Squiz.Commenting.FileComment.Missing)
-----------------------------------------------------------------------------

Time: 623ms; Memory: 12MB
```

Seriously, please do some debugging yourself before reporting an issue and report it with a proper reproduction case.
I've seen so many issues by you already and only 1 in 20 is valid.

**Comment by jrfnl** [comment]
@kkmuffme There might be something in your dependencies which is incompatible with PHP 8.0 (used during `composer install`) which is causing a fatal error when the autoload file is requested on PHP 8.0 (which is needed to run _any_ plugins).

Also see: https://github.com/PHPCSStandards/composer-installer/issues/79

Having said that, if that's the case, than:
1. You still never provided a valid reproduction case. Purely based on the info provided, this issue cannot be reproduced.
2. It is a not a problem with this plugin, but a problem with how you run Composer.
    The `platform.php` setting is not intended to allow you to use a _low_ PHP version with a _high_ platform setting.. It is intended for allowing to run Composer on a higher version and prevent locking in dependencies not compatible with your _minimum_ PHP version.

**Comment by jrfnl** [comment]
> 1. I can reproduce it with the exact steps outlined above. Since the workaround is easy enough, I don't want to waste your and my time.

No you can't. There must be something else in your `composer.json` which is causing the problem. And without that extra information it's a guessing game. (which, yes, has wasted my time)

>     2. That's wrong, see the example from composer's docs :-) (lower PHP version with higher platform setting)

See my previous answer. You can use it like that, but if _anything_ in your dependencies will have installed something which includes syntax incompatible with the _runtime_ PHP version for Composer, the `autoload.php` file will throw a fatal and will block all plugins from working.

In other words, know your tools.
